### PR TITLE
Update to the latest build daemon

### PR DESCRIPTION
**Pull request by jakemac53**
Primarily this is updating to the new ServerLog api.

I added git overrides for now but we could wait to merge this until publishing the build_daemon/build_runner packages, and so it can merge cleanly without overrides.

### CODE REVIEW

File: webdev/lib/src/logging.dart [hunk 1]
```diff
@@ -5,77 +5,84 @@
 import 'dart:io';
 
 import 'package:io/ansi.dart';
-import 'package:build_daemon/data/server_log.dart';
 import 'package:logging/logging.dart';
 
-var _verbose = false;
+typedef LogHandler = void Function(Level level, String message,
```
Comment (grouma) [inline webdev/lib/src/logging.dart]: This is slightly confusing as the build daemon client exposes a logHandler argument with a different type. Maybe call this WebdevLogHandler?

Comment (jakemac53) [reply webdev/lib/src/logging.dart]: its not public or anything, and this is in package:webdev after all.... plus it has a capital L so totally different :P 

I will consider it though haha

Comment (grouma) [reply webdev/lib/src/logging.dart]: Renamed it to `logWriter` as to not confuse it with the `logHandler`.

File: example/pubspec.yaml [hunk 2]
```diff
@@ -7,3 +7,13 @@ environment:
 dev_dependencies:
   build_runner: '>=1.3.0 <2.0.0'
   build_web_compilers: '>=1.0.0 <3.0.0'
+
+dependency_overrides:
+  build_daemon:
```
Comment (jakemac53) [inline example/pubspec.yaml]: this one can probably be removed now (its published)

File: webdev/pubspec.yaml [hunk 3]
```diff
@@ -50,5 +50,15 @@ dev_dependencies:
   uuid: ^2.0.0
   webdriver: ^2.0.0
 
+dependency_overrides:
+  build_daemon:
```
Comment (jakemac53) [inline webdev/pubspec.yaml]: same this can be removed now

**Pull request merged**